### PR TITLE
Add more instruction and add docker binary support in docker/init.sls

### PR DESCRIPTION
**Pull request by Random-Liu**
I met some problem with updating cluster docker version when validating docker 1.10.

In this PR, I updated the comments and add the docker binary support, hope this will make the validation easier in the future.

### CODE REVIEW

File: cluster/saltbase/salt/docker/init.sls [hunk 1]
```diff
@@ -195,9 +197,22 @@ net.ipv4.ip_forward:
...
+#        docker_pkg_name - docker-engine
+#        override_deb - docker-engine_1.8.3-0~wheezy_amd64.deb
+#        override_deb_sha1 - f812e1134e7375b29886a372a6e24d04e1ee33c9
+#        override_docker_ver - 1.8.3-0~wheezy
+# 7. Sometimes debian package may not work, to use static linked binary version:
+#    * Find url of specific binary version at:
+#        https://github.com/docker/docker/releases
+#    * Download docker binary and upload to GCS, just like what we do for debian package.
+#    * Update override_bin_name with new binary name. For example, for docker 1.8.3:
+#        override_bin_url - docker-1.8.3
```
Comment (yujuhong) [inline cluster/saltbase/salt/docker/init.sls]: s/override_bin_url /override_bin_name

Comment (yujuhong) [inline cluster/saltbase/salt/docker/init.sls]: If both override_deb and override_bin_name are set, the latter will be used. Is that correct?

Comment (Random-Liu) [inline cluster/saltbase/salt/docker/init.sls]: ACK. :)

**Comment by Random-Liu** [comment]
Addressed comment and squashed.

File: cluster/saltbase/salt/docker/init.sls [hunk 2]
```diff
@@ -195,9 +197,23 @@ net.ipv4.ip_forward:
 #    shasum <deb>
 # 6. Update override_deb, override_deb_sha1, override_docker_ver with new
 #    deb name, new hash and new version
+#    For example, for docker 1.8.3-0~wheezy_amd64.deb:
+#        docker_pkg_name - docker-engine
+#        override_deb - docker-engine_1.8.3-0~wheezy_amd64.deb
+#        override_deb_sha1 - f812e1134e7375b29886a372a6e24d04e1ee33c9
+#        override_docker_ver - 1.8.3-0~wheezy
+# 7. Sometimes debian package may not work, to use static linked binary version:
```
Comment (dchen1107) [inline cluster/saltbase/salt/docker/init.sls]: I am not sure we want to have extra complexity here in our salt file. It is fragile already, and when we hit that issue, it is docker p0 issue, and they are fixing it pretty fast. We could document this in a separate document or wiki page.

**Comment by dchen1107** [comment]
Beside above comment, looks good to me.

File: cluster/saltbase/salt/docker/init.sls (hunk 2 above)
Comment (Random-Liu) [inline cluster/saltbase/salt/docker/init.sls]: Make sense.
I thought the binary package is another "official way" of docker deployment. I'll remove this and keep it in my note. :)

**Comment by yujuhong** [comment]
Is this PR still active?

**Comment by Random-Liu** [comment]
@yujuhong No, @dchen1107 asked for the script for docker validation just now, :)
I just update the PR for her. Will close this later~

**Comment by dchen1107** [comment]
cc/ @timstclair

**Pull request closed without merging**